pkg/upgrade/handle_batch.go

```diff
@@ -94,6 +95,10 @@ func (ctl *Controller) handleJobs(ctx context.Context) error {
 			}
 			return obj, enqueueOrDelete(jobs, obj, upgradejob.ConditionComplete)
 		}
```

Comment:
Would like a comment here, as with the other blocks, explaining what we are deciding to do and why, e.g.:
> // if the job is hasn't failed or completed but is not on the applying list, consider it running out-of-turn and delete it